test/helpers/index.ts

```diff
@@ -0,0 +1,49 @@
+import { btc2sat } from "src/utils"
+import { FtxDealerWallet } from "src/dealer/FtxDealerWallet"
+import {
+  balanceSheetIsBalanced,
+  updateUsersPendingPayment,
+} from "src/ledger/balanceSheet"
+import { waitUntilChannelBalanceSyncAll } from "./lightning"
+
+export * from "./bitcoinCore"
+export * from "./lightning"
+export * from "./user"
+
+export const amountAfterFeeDeduction = ({ amount, depositFeeRatio }) =>
+  Math.round(btc2sat(amount) * (1 - depositFeeRatio))
+
+export const checkIsBalanced = async () => {
+  await updateUsersPendingPayment()
+  // wait for balance updates because invoice event
+  // arrives before wallet balances updates in lnd
+  await waitUntilChannelBalanceSyncAll()
+
+  const { assetsLiabilitiesDifference, bookingVersusRealWorldAssets } =
+    await balanceSheetIsBalanced()
+  expect(assetsLiabilitiesDifference).toBeFalsy() // should be 0
+
+  // FIXME: because safe_fees is doing rounding to the value up
+  // balance doesn't match any longer. need to go from sats to msats to properly account for every msats spent
+  expect(Math.abs(bookingVersusRealWorldAssets)).toBeLessThanOrEqual(5) // should be 0
```

Comment:
maybe this could be 1 or 0 not that there is no variable routing fees?

the reason this was here is because lnd switch it's primary unit of account from `sats` to `msats` and because we only managed sats as of now in our books, we were doing rounding errors and a result the books were no longer accurate.

if we charge a routing fee of 1 sats/tx (is it 1 sats or 1 msats?) then we can reduce the error threshold here.

Reply:
it is [1 sat](https://github.com/dolcalmi/galoy/blob/refactor_test_stg3a/test/helpers/lightning.ts#L191). I will run the tests changing the threshold

Reply:
Fixed https://github.com/GaloyMoney/galoy/pull/340/commits/25087ca130b49e79a9a0cfe9ea94279b1164b68b now run with 0 but I had to add some sleeps to cancel hodl invoice